test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -917,6 +917,152 @@ public void PackCommand_PackProject_OutputsContentFilesInNuspecForMultipleFramew
             }
         }
 
+        [Platform(Platform.Windows)]
+        [Fact]
+        public void PackCommand_SingleFramework_GeneratesPackageOnBuild()
+        {
+            using (var testDirectory = TestDirectory.Create())
+            {
+                var projectName = "ClassLibrary1";
+                var workingDirectory = Path.Combine(testDirectory, projectName);
+                var projectFileContents =
+                    $@"<Project Sdk=""Microsoft.NET.Sdk"" ToolsVersion=""15.0"">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard1.4</TargetFramework>
+    <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include=""**\*.cs"" />
+    <EmbeddedResource Include=""**\*.resx"" />
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include=""NETStandard.Library"" Version=""1.6"" />
+  </ItemGroup>
+
+</Project>";
```

Comment:
If these templates can be reused then maybe have them as files like - https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Test.Utility/compiler/resources/project2.csproj